File: src/main/java/me/miki/shindo/utils/file/DirectoryUtils.java
package me.miki.shindo.utils.file;

import java.io.File;

public class DirectoryUtils {

    public static void deleteDirectory(File directory) {
        if (directory.exists()) {
            File[] files = directory.listFiles();

            if (files != null) {
                for (File file : files) {
                    if (file.isDirectory()) {
                        deleteDirectory(file);
                    } else {
                        file.delete();
                    }
                }
            }

            directory.delete();
        }
    }

    public static long getDirectorySize(File directory) {

        long size = 0;

        if (directory.isDirectory()) {
            File[] files = directory.listFiles();

            if (files != null) {
                for (File file : files) {
                    if (file.isFile()) {
                        size += file.length();
                    } else if (file.isDirectory()) {
                        size += getDirectorySize(file);
                    }
                }
            }
        } else if (directory.isFile()) {
            size = directory.length();
        }

        return size;
    }
}

File: src/main/kotlin/me/miki/shindo/utils/file/DirectoryUtils.kt
package me.miki.shindo.utils.file

import java.io.File

object DirectoryUtils {
    @JvmStatic
    fun deleteDirectory(directory: File) {
        if (directory.exists()) {
            val files = directory.listFiles()
            if (files != null) {
                for (file in files) {
                    if (file.isDirectory) {
                        deleteDirectory(file)
                    } else {
                        file.delete()
                    }
                }
            }
            directory.delete()
        }
    }

    @JvmStatic
    fun getDirectorySize(directory: File): Long {
        var size: Long = 0
        if (directory.isDirectory) {
            val files = directory.listFiles()
            if (files != null) {
                for (file in files) {
                    size +=
                        if (file.isFile) {
                            file.length()
                        } else if (file.isDirectory) {
                            getDirectorySize(file)
                        } else {
                            0
                        }
                }
            }
        } else if (directory.isFile) {
            size = directory.length()
        }
        return size
    }
}
